### ISSUE TITLE
Can't it handle parameter?

### ISSUE THREAD

**Issue by rottmann** [nchutchind/vue-outside-events#2]
e.g.
```
<div :v-mouseup-outside="onMouseUp(col)"></div>
```

**Comment by nchutchind** [comment]
Not currently. The directive currently takes a function reference to call when the event happens, and it sends that function the event that occurred and the element that triggered the event (ie, you were listening for a click outside element #A and the user clicked on element #B, which is outside #A, then the function would get a click event and a reference to #B).

The only way that I know of to pass additional parameters would be to change the directive to optionally receive an object that had a "callback" property (to replace the function that would normally have been provided) and would call the callback with the event, the triggering element, and the object that was passed in. I can't currently think of another way to do this that would still be generic, though I welcome any suggestions. Note: there are ways for the directive to access Vue parameters that have been placed on the element, but they would be named and it would feel like the directive was leaking, to me.

I'll try this out, but would gladly look at a pull request as well.

**Comment by nchutchind** [comment]
One thing I forgot to mention was that Directive binding happens only once, at run-time. Directives in Vue do not get updated when data values change, only when the DOM element or element's children get updated. This is a quality of the Vue library which you can see [here](https://vuejs.org/v2/guide/custom-directive.html).

**Comment by nchutchind** [comment]
This should generally be solved with version 1.1.0 of the plugin. There is a demo page located [here](https://nchutchind.github.io/vue-outside-events/docs/demo-pass-object.html) and you can generally see how the code works below. Just remember, because Directives are only bound once, they will only have the initial data values at bind-time. The bound values do not update because the directive does not update.

`<div v-click-outside="{ handler: onClickOutside, foo: someVar }" />`

```
new Vue({
    data: {
        someVar: "FOO"
    },
    methods: {
        onClickOutside(e, el, extras) {
            console.log(extras.foo);
        }
    }
}
```
